functions/Export-DbaDiagnosticQuery.ps1

```diff
@@ -0,0 +1,140 @@
+Copyright (C) 2016 Chrissy LeMaire
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.	
+
+.LINK
+https://dbatools.io/Export-DbaDiagnosticQuery
+
+.EXAMPLE  
+Invoke-DbaDiagnosticQuery -SqlInstance sql2016 | Export-DbaDiagnosticQuery -Path c:\temp
+
+Converts output from Invoke-DbaDiagnosticQuery to multiple CSV files
+	
+.EXAMPLE 
+$output = Invoke-DbaDiagnosticQuery -SqlInstance sql2016
+Export-DbaDiagnosticQuery -InputObject $output -ConvertTo Excel
+
+Converts output from Invoke-DbaDiagnosticQuery to Excel worksheet(s) in the Documents folder
+
+#>
+	[CmdletBinding()]
+	param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[object[]]$InputObject,
+		[ValidateSet("Excel", "Csv")]
+		[string]$ConvertTo = "Csv",
+		[System.IO.FileInfo]$Path = [Environment]::GetFolderPath("mydocuments"),
+		[string]$Suffix = "$(Get-Date -format 'yyyyMMddHHmmssms')",
+		[switch]$Silent
+	)
+	
+	begin {
+		if ($convertto -eq "Excel") {
+			try {
+				Import-Module ImportExcel -ErrorAction Stop
+			}
+			catch {
+				Stop-Function -Message "Failed to load module, exporting to Excel feature is not available"
+				Stop-Function -Message "Install the module from: https://github.com/dfinke/ImportExcel"
+				Stop-Function -Message "Valid alternative conversion format is csv"
+			}
+		}
+		
+		Function Remove-InvalidFileNameChars {
+			param (
+				[Parameter(Mandatory = $true,
+						   Position = 0,
+						   ValueFromPipeline = $true,
+						   ValueFromPipelineByPropertyName = $true)]
+				[String]$Name
+			)
+			$Name = $Name.Replace(" ","-")
+			$invalidChars = [IO.Path]::GetInvalidFileNameChars() -join ''
+			$re = "[{0}]" -f [RegEx]::Escape($invalidChars)
+			return ($Name -replace $re)
+		}
+	}
+	
+	process {
+		if (Test-FunctionInterrupt) { return }
+		
+		foreach ($row in $inputobject) {
+			$results = $row.Result
+			$name = $row.Name
+			$sqlserver = $row.SqlInstance.Replace("\", "$")
+			$dbname = $row.DatabaseName
+			$number = $row.Number
+			
+			if ($null -eq $results){
+				Stop-Function -Message "Resultset was empty for $name" -Target $row -Continue
+			}
+			
+			foreach ($result in $results) {
+				
+				if ($null -eq $result) {
+					Stop-Function -Message "Result was empty for $name" -Target $row -Continue
```

Comment:
This will be missleading: It reports for the entire row, but only reflect a single result within that row.
Also note that this will continue with the next result, if `$results` has more than one, empty or not, rather than moving to the next row. If that's intentional: Fine. If not: You'll need to label your loops.